src/ripple/rpc/impl/KeypairForSignature.cpp

```diff
@@ -65,31 +65,43 @@ std::pair<PublicKey, SecretKey>
 keypairForSignature (Json::Value const& params, Json::Value& error)
 {
     bool const has_key_type  = params.isMember (jss::key_type);
-    bool const hasPassphrase = params.isMember (jss::passphrase);
-    bool const hasSecret     = params.isMember (jss::secret);
-    bool const hasSeed       = params.isMember (jss::seed);
-    bool const hasSeedHex    = params.isMember (jss::seed_hex);
 
-    int const n_secrets =
-        (hasPassphrase ? 1 : 0) +
-        (hasSecret ? 1 : 0) +
-        (hasSeed ? 1 : 0) +
-        (hasSeedHex ? 1 : 0);
+    // All of the signature types we allow.  Only one, though!
+    static constexpr char const* const sigTypes[]
+    {
+        jss::passphrase.c_str(),
+        jss::secret.c_str(),
+        jss::seed.c_str(),
+        jss::seed_hex.c_str()
+    };
+
+    // Identify which signature type is in use.
+    constexpr char noSig[] = "no signature";
+    char const* sigType = noSig;
+    int sigTypeCount = 0;
+    for (auto t : sigTypes)
+    {
+        if (params.isMember (t))
+        {
+            ++sigTypeCount;
+            sigType = t;
+        }
+    }
 
-    if (n_secrets == 0)
+    if (sigTypeCount == 0 || sigType == noSig)
```

Comment:
I see your point about dereferencing a `nullptr`. I just don't like magic strings / numbers.

In addition to `nullptr`, I can think of two alternatives to consider.
1) `boost::optional<char const*> sigType`. This line still checks for `!sigType`, but you need `*sigType` below where it's used as a `char*`.
2) A little bit more flight of fancy, but replace `sigType` and `sigTypeCount` both with `std::vector<char const*> sigType`. The increment / assignment becomes an `emplace_back`. The other comparisons become `sigType.size()` and `sigType.front()`.